### PR TITLE
Issue #7105: remove deprecated constructor from XMLLogger

### PR DESCRIPTION
**Pull request by rnveach**
Issue #7105

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java [hunk 1]
```diff
@@ -1323,7 +1328,7 @@ public void testDefaultLoggerClosesItStreams() throws Exception {
                 new CloseAndFlushTestByteArrayOutputStream()) {
             checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
             checker.addListener(new DefaultLogger(testInfoOutputStream,
-                true, testErrorOutputStream, true));
```
Comment (pbludov) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: Should  this be done in separate issue? The ctor for `DefaultLogger` is not affected by this PR.

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java [hunk 2]
```diff
@@ -1379,8 +1384,8 @@ public void testDuplicatedModule() throws Exception {
         // BriefUtLogger does not print the module name or id postfix,
         // so we need to set logger manually
         final ByteArrayOutputStream out = Whitebox.getInternalState(this, "stream");
-        final DefaultLogger logger =
-                new DefaultLogger(out, true, out, false, new AuditEventDefaultFormatter());
```
Comment (pbludov) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: `DefaultLogger` with four parameters again. See my previous comment.

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java (hunk 1 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: Done.

Splitting up previous 1 PR into many. Missed that this file contained multiple fixes in 1.

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java (hunk 2 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: Done.

**Pull request merged**